service/history/ndc/workflow_state_replicator.go

```diff
@@ -279,29 +280,36 @@ func (r *WorkflowStateReplicatorImpl) backfillHistory(
 	namespaceID namespace.ID,
 	workflowID string,
 	runID string,
+	rootRunID string,
```

Comment:
nit: maybe rename to originalRunID here. rootRunID concept is used in other place for parent/child relationship.

Reply:
Will update the name